src/Raven.Client/Documents/Operations/AI/AiUsage.cs

```diff
@@ -35,17 +44,20 @@ public DynamicJsonValue ToJson()
             [nameof(CompletionTokens)] = CompletionTokens,
             [nameof(TotalTokens)] = TotalTokens,
             [nameof(CachedTokens)] = CachedTokens,
+            [nameof(ReasoningTokens)] = ReasoningTokens
         };
     }
 
     internal static AiUsage GetUsageDifference(AiUsage usage2, AiUsage usage1)
     {
+        var lastTotalWithoutReasoning = (usage1.CompletionTokens + usage1.PromptTokens);
         return new AiUsage
         {
-            PromptTokens = usage2.PromptTokens - usage1.PromptTokens,
-            TotalTokens = usage2.TotalTokens - usage1.TotalTokens,
+            PromptTokens = usage2.PromptTokens - lastTotalWithoutReasoning,
+            TotalTokens = usage2.TotalTokens - lastTotalWithoutReasoning >= 0 ? usage2.TotalTokens - lastTotalWithoutReasoning : usage2.TotalTokens, //should not be negative but just to make sure we have no negative values
```

Comment:
No, here the risk is that you'll have `usage2.TotalTokens - lastTotalWithoutReasoning` being negative.